cmd/dep/root_analyzer.go

```diff
@@ -167,9 +168,23 @@ func (a *rootAnalyzer) DeriveManifestAndLock(dir string, pr gps.ProjectRoot) (gp
 func (a *rootAnalyzer) FinalizeRootManifestAndLock(m *dep.Manifest, l *dep.Lock, ol dep.Lock) {
 	// Iterate through the new projects in solved lock and add them to manifest
 	// if they are direct deps and log feedback for all the new projects.
+	//
+	// IDEA: Make a lookup map of ProjectRoot ->projects in the old lock
+	// use that data structure to simplify these loops
+	olLookup := make(map[gps.ProjectRoot]gps.LockedProject)
+	for _, op := range ol.Projects() {
+		olLookup[op.Ident().ProjectRoot] = op
+	}
 	for _, y := range l.Projects() {
 		var f *fb.ConstraintFeedback
 		pr := y.Ident().ProjectRoot
+
+		op, inOl := olLookup[pr]
+		if inOl {
+			if diff := gps.DiffProjects(y, op); diff != nil {
+				fmt.Printf("Some changes: \nImported: %v\nRevised: %v\nDiff: %+v\n", y, op, diff)
```

Comment:
Let's go with your suggestion and add a new method to the feedback package like `NewBrokenImportFeedback` which prints something like this:

```
Warning: Unable to preserve imported lock v1.2.0 (abc123) for github.com/example/A. Locking in v1.2.1 (def456).
```

Reply:
When you add a new feedback function, lets make sure to also add a unit test for it. There should be other tests in feedback_test that you can use as an example.